### PR TITLE
Refactor postcondition for class initialization

### PR DESCRIPTION
**Pull request by khasizadaj**
In order to have a clear postcondition, I added a new class method (`__is_reset`) which performs the same checks as before.

I updated `not` statements as well, in order to have more readable code.

### CODE REVIEW

File: src/readonce.py [hunk 1]
```diff
@@ -60,6 +60,10 @@ def __reset_is_consumed(cls):
     def __update_is_consumed(cls):
         cls.__is_consumed = True
 
+    @classmethod
+    def __is_reset(cls):
+        return len(cls.__secrets) == 0 and cls.__is_consumed is False
```
Comment (ShahriyarR) [inline src/readonce.py]: The idea of simple function is great and neat thanks for your time.
But, I would like to leave as `not cls.secrets and not cls.__is_consumed`
Could you please kindly change this one?
Thanks.

**Review by ShahriyarR — CHANGES_REQUESTED** [review]

File: src/readonce.py (hunk 1 above)
Comment (khasizadaj) [reply src/readonce.py]: Changed the conditions. Please, see in the next commit.

**Comment by ShahriyarR** [comment]
@khasizadaj the test run failed with:

```py
pytest -svv
ImportError while loading conftest '/home/runner/work/py-read-once/py-read-once/tests/conftest.py'.
tests/conftest.py:78: in <module>
    class DBCredentialsWithDescriptors:
tests/conftest.py:79: in DBCredentialsWithDescriptors
    password: Password = Password("db-password")
/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/icontract/_checkers.py:641: in wrapper
    result = func(*args, **kwargs)
tests/conftest.py:29: in __init__
    super().__init__()
/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/icontract/_checkers.py:810: in wrapper
    result = func(*args, **kwargs)
/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/icontract/_checkers.py:646: in wrapper
    violation_error = _assert_postconditions(
/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/icontract/_checkers.py:396: in _assert_postconditions
    exception = _create_violation_error(contract=contract, resolved_kwargs=resolved_kwargs)
/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/icontract/_checkers.py:192: in _create_violation_error
    raise RuntimeError(''.join(parts)) from err
E   RuntimeError: Failed to recompute the values of the contract condition:
E   File /opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/readonce.py, line 46 in ReadOnce:
E   lambda self: self.__is_reset()
make: *** [Makefile:28: test] Error 4
Error: Process completed with exit code 2.
```

Did you run `make test` locally and check it?

**Comment by khasizadaj** [comment]
I did check it, but couldn't solve the issue. For some reason, `icontract` do not accept `__is_reset` method. I couldn't find any solution for it. Any ideas?

File: src/readonce.py [hunk 2]
```diff
@@ -60,6 +60,10 @@ def __reset_is_consumed(cls):
     def __update_is_consumed(cls):
         cls.__is_consumed = True
 
+    @classmethod
+    def __is_reset(cls):
+        return not cls.secrets and not cls.__is_consumed
```
Comment (ShahriyarR) [inline src/readonce.py:65]: Maybe this is due to `cls.secrets` call? It should be `cls.__secrets`?

**Review by ShahriyarR — CHANGES_REQUESTED** [review]

**Pull request closed without merging**